packages/mui-material/src/Backdrop/Backdrop.js

```diff
@@ -67,22 +66,34 @@ const Backdrop = React.forwardRef(function Backdrop(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const rootSlotProps = slotProps.root ?? componentsProps.root;
+  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };
+
+  const externalForwardedProps = {
+    slots: backwardCompatibleSlots,
+    slotProps,
+  };
```

Comment:
Hey @harry-whorlow! We shouldn't remove `components` and `componentsProps`. We should deprecate them **in a separate PR** after this one. For this PR we should do the following:

```suggestion
  const backwardCompatibleSlots = { transition: TransitionComponentProp, root: components.Root, ...slots };
  const backwardCompatibleSlotProps = { ...componentsProps, ...slotProps };

  const externalForwardedProps = {
    slots: backwardCompatibleSlots,
    slotProps: backwardCompatibleSlotProps,
  };
```

Reply:
ah... yeah, good point!

and thanks for the code suggestion